junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertThrows.java

```diff
@@ -60,15 +58,18 @@ private static <T extends Throwable> T assertThrows(Class<T> expectedType, Execu
 			}
 			else {
 				UnrecoverableExceptions.rethrowIfUnrecoverable(actualException);
-				String message = buildPrefix(nullSafeGet(messageOrSupplier))
-						+ format(expectedType, actualException.getClass(), "Unexpected exception type thrown");
-				throw new AssertionFailedError(message, actualException);
+				throw assertionFailure() //
+						.message(messageOrSupplier) //
+						.expected(expectedType) //
+						.actual(actualException.getClass()) //
+						.reason("Unexpected exception type thrown") //
+						.build();
```

Comment:
Shouldn't `actualException` be handed over using `cause(actualException)` or was it intended to omit it here?